front/src/modules/trainschedule/components/ManageTrainSchedule/helpers/handleParseFiles.ts

```diff
@@ -9,71 +9,40 @@ export const handleFileReadingError = (error: Error) => {
   console.error('File reading error:', error);
 };
 
-export const handleJsonParsingError = (error: Error, dispatch: Dispatch) => {
-  console.error('Error parsing JSON:', error);
-  dispatch(
-    setFailure({
-      name: t('errorMessages.error'),
-      message: t('errorMessages.errorInvalidJSONFormat'),
-    })
-  );
-};
-
-export const handleXmlParsingError = (error: Error, dispatch: Dispatch) => {
-  console.error('Error parsing XML/RailML:', error);
-  dispatch(
-    setFailure({
-      name: t('errorMessages.error'),
-      message: t('errorMessages.errorInvalidXMLFormat'),
-    })
-  );
-};
-
 export const processJsonFile = (
   fileContent: string,
   setTrainsJsonData: (data: TrainScheduleBase[]) => void,
-  dispatch: Dispatch
+  dispatch: Dispatch,
+  t: TFunction
 ) => {
-  try {
-    const importedTrainSchedules: TrainScheduleBase[] = JSON.parse(fileContent);
-    if (importedTrainSchedules && importedTrainSchedules.length > 0) {
-      setTrainsJsonData(importedTrainSchedules);
-    }
-  } catch (error) {
-    handleJsonParsingError(error as Error, dispatch);
+  const importedTrainSchedules: TrainScheduleBase[] = JSON.parse(fileContent);
+  if (importedTrainSchedules && importedTrainSchedules.length > 0) {
```

Comment:
On the line above, we type `importedTrainSchedules: TrainScheduleBase[]` so we shouldn't have to check for that ?
```suggestion
  if (importedTrainSchedules.length > 0) {
```

Reply:
Yes I can add a check :)